src/lex.js

```diff
@@ -146,7 +149,7 @@ module.exports = function(script) {
                 }
             }
 
-            tokens.push(new types.IntValue(digits));
+            tokens.push(new tokenTypes.Literal('integer', digits));
```

Comment:
Hmm, ideally you should make it into an IntValue for the token, otherwise bounds checks won't happen. And it's just simpler to convert here anyway. Perhaps this?

    tokens.push(new tokenTypes.Literal(new types.IntValue(digits));

Though maybe I'm going about this wrong